### PR TITLE
Gradient Images

### PR DESCRIPTION
**Pull request by LucaHaverty**
### Description
Adds support for images to have gradients, and updates all dynamic images that will have a gradient to this new system. If an image is created that doesn't have a sprite, a gradient updater component will be added to it that will make a new material for it from a custom shader that handles the gradient as well as rounding the corners. Any images that have a sprite assigned to them or have a sprite added will not have this component.

*Note: this goal of this pr is not to get all ui elements that use gradient to look exactly like the designs on figma, just to get gradients to work* 

### Objectives
- [x] Gradient shader that also rounds corners
- [x] Update all prefabs that will have gradients to use the new shader

### Testing Done
- Looked around at various ui to make sure nothing had gradients that was not supposed to
- Scaled and positioned objects in various ways to make sure the gradients were updating

[JIRA Issue](https://jira.autodesk.com/browse/AARD-1571)

### CODE REVIEW

File: engine/Assets/Materials/Shaders/ImageGradient/RoundedCorners.shader
Comment (BrandonPacewic) [inline engine/Assets/Materials/Shaders/ImageGradient/RoundedCorners.shader:1]: Can we clean this file up? Looks like there is some stuff here that is not being used. Let me know if that's incorrect I'm not deeply familiar with Unity shader code.

File: engine/Assets/Scripts/UI/Dynamic/DynamicUIComponents.cs [hunk 1]
```diff
@@ -1180,7 +1197,12 @@ public class Dropdown : UIComponent {
             _itemLabel.SetColor(ColorManager.SynthesisColor.InteractiveElementText);
 
             _viewportImage = new Image(this, unityObject.transform.Find("Template").Find("Viewport").gameObject);
-            _viewportImage.SetColor(ColorManager.SynthesisColor.InteractiveElement);
+
+            _viewportImage.SetColor(ColorManager.SynthesisColor.InteractiveElementRight,
+                ColorManager.SynthesisColor.InteractiveElementLeft);
+
+            // TODO: Get some more control over the individual items in the dropdown
```
Comment (BrandonPacewic) [inline engine/Assets/Scripts/UI/Dynamic/DynamicUIComponents.cs:1204]: Leaving this `TODO`?

File: engine/Assets/Scripts/Modes/PracticeMode.cs [hunk 2]
```diff
@@ -139,16 +139,17 @@ public class PracticeMode : IMode {
             _showingScoreboard = true;
             DynamicUIManager.CreatePanel<ScoreboardPanel>(true, false);
         }
-        bool openEscapeMenu = InputManager.MappedValueInputs[TOGGLE_ESCAPE_MENU_INPUT].Value == 1.0F;
+        // TODO: Fix this error
```
Comment (BrandonPacewic) [inline engine/Assets/Scripts/Modes/PracticeMode.cs:142]: Is this important?

**Review by BrandonPacewic — CHANGES_REQUESTED** [review]
Looks like everything seems to be working.

File: engine/Assets/Scripts/Modes/PracticeMode.cs (hunk 2 above)
Comment (LucaHaverty) [reply engine/Assets/Scripts/Modes/PracticeMode.cs:142]: Oh yeah, I was getting the escape button error again. I will fix that

File: engine/Assets/Materials/Shaders/ImageGradient/RoundedCorners.shader
Comment (LucaHaverty) [reply engine/Assets/Materials/Shaders/ImageGradient/RoundedCorners.shader:1]: All the commented out code was from me attempting to get unity UI masks to work with the shader, but I was never able to. I'll remove it. The only time we used UI masks was the dropdown but we didn't need to.

Comment (BrandonPacewic) [reply engine/Assets/Materials/Shaders/ImageGradient/RoundedCorners.shader:1]: 👍

File: engine/Assets/Scripts/UI/Dynamic/DynamicUIComponents.cs (hunk 1 above)
Comment (LucaHaverty) [reply engine/Assets/Scripts/UI/Dynamic/DynamicUIComponents.cs]: Removed

**Review by KyroVibe — CHANGES_REQUESTED** [review]
Couple of things I've found:
Screenshot 2023-07-17 at 2 35 16 PM
The input field when making a new theme is a gradient, Idk if thats intentional or not.

Screenshot 2023-07-17 at 2 33 46 PM

The fields in the drop downs are masked correctly, but the icon for indicating which one is selected instead. This will shift to a kinda grey out as seen in the UI mocks, but still I'd like this masking issue either fixed or determined why it's happening so we can avoid it in the future.

I've also noticed the scrollbars just kinda seem to be gone as well.

**Comment by a-crowell** [comment]
Looks good in the Unity Editor. After building it into an executable though, little of the UI works with a bunch of shader being null errors. How can I get the assets to make it work?
image

**Comment by KyroVibe** [comment]
> Looks good in the Unity Editor. After building it into an executable though, little of the UI works with a bunch of shader being null errors. How can I get the assets to make it work?
> image
> 
> 

I can't remember off the top of my head, but there is a list of all included shaders in the executables, and it needs to be added to that. I believe it's under graphics settings

**Comment by a-crowell** [comment]
> > Looks good in the Unity Editor. After building it into an executable though, little of the UI works with a bunch of shader being null errors. How can I get the assets to make it work?
> > image
> 
> I can't remember off the top of my head, but there is a list of all included shaders in the executables, and it needs to be added to that. I believe it's under graphics settings

Gotcha. If you've tested that, then sounds good to me.

**Comment by a-crowell** [comment]
Is the parent object button going to be gradient?
image

**Comment by LucaHaverty** [comment]
> Is the parent object button going to be gradient? image

Not sure. Probably

**Comment by LucaHaverty** [comment]
> I've also noticed the scrollbars just kinda seem to be gone as well.

@KyroVibe The scrollbars on the dropdown?

**Comment by KyroVibe** [comment]
> > I've also noticed the scrollbars just kinda seem to be gone as well.
> 
> @KyroVibe The scrollbars on the dropdown?

Thats so far the only place I've seen it

**Comment by LucaHaverty** [comment]
@KyroVibe The shader should now behave normally with masks

**Review by Markytt — COMMENTED** [review]
In the score zone panel, "Red Alliance" has a blue background.

Screenshot 2023-07-18 104014

**Review by PepperLola — CHANGES_REQUESTED** [review]
It looks good, except now the match mode robot placement buttons don't have any color, and both alliance buttons are blue as @Markytt mentioned earlier. 
Screenshot 2023-07-18 at 10 45 02 AM

**Review by PepperLola — APPROVED** [review]
Looks good!

**Review by KyroVibe — APPROVED** [review]
Masks are fixed, nice work

**Pull request merged**